src/compressed_tensors/quantization/observers/min_max.py

```diff
@@ -45,21 +45,62 @@ def calculate_qparams(self, observed: Tensor) -> Tuple[FloatTensor, IntTensor]:
         """
         # TODO: Add support for full range of quantization Args, only supports 8bit
         #       per tensor
-        min_val = torch.tensor([observed.min()])
-        max_val = torch.tensor([observed.max()])
 
-        # update running average
-        if self.counter > 0:
-            self.min_val = (self.min_val * self.counter + min_val) / (self.counter + 1)
-            self.max_val = (self.max_val * self.counter + max_val) / (self.counter + 1)
+        # channel wise quantization
+        if self.quantization_args.group_size == -1 and observed.dim() > 2:
+            scale_zero_points = []
+            for chan_i in range(observed.shape[0]):
+
+                min_val = torch.tensor([observed[chan_i:].min()])
+                max_val = torch.tensor([observed[chan_i:].max()])
+
+                # update running average
+                if self.counter > 0:
+                    self.min_val = (self.min_val * self.counter + min_val) / (
+                        self.counter + 1
+                    )
+                    self.max_val = (self.max_val * self.counter + max_val) / (
+                        self.counter + 1
+                    )
+                else:
+                    self.min_val = min_val
+                    self.max_val = max_val
```

Comment:
see per-token draft PR #24 - we can avoid looping here by reducing everything but the channel axis (ie using `torch.amin` and `torch.amax`)